src/Components/MainTableView/UpdateTuple.tsx

```diff
@@ -0,0 +1,510 @@
+          this.setState({errorMessage: 'Missing require field: ' + secondaryAttribute.attributeName});
+          return;
+        }
+      }
+    }
+
+    // All checks passed thus attempt insert
+    fetch('/api/update_tuple', {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json', 'Authorization': 'Bearer ' + this.props.token},
+      body: JSON.stringify({schemaName: this.props.selectedSchemaName, tableName: this.props.selectedTableName, tuple: tupleBuffer})
+    })
+    .then(result => {
+      // Check for error mesage 500, if so throw and error
+      if (result.status === 500) {
+        result.text()
+        .then(errorMessage => {throw new Error(errorMessage)})
+        .catch(error => {
+          this.setState({errorMessage: error.message});
+        });
+      }
+      return result.text();
+    })
+    .then(result => {
+      // clear selection and buffer upon successful delete
+      this.props.clearEntrySelection();
+      this.setState({tupleBuffer: {}})
+
+      // Update was sucessful, tell TableView to fetch the content again
+      this.props.fetchTableContent();
+    })
+    .catch((error) => {
+      this.setState({errorMessage: error.message});
+    })
+  }
+
+  /**
+   * Function dealing with when user clicks on the reset icon for nullable input field. 
+   * TODO: Align behavior with the edge case specs - whether to null, or fill with default
+   * @param tableAttribute Table attribute object so the function can extract the attributeName 
+   */
+  resetToNull(tableAttribute: any) {
+    if (Object.entries(this.state.tupleBuffer).length) {
+      let updatedBuffer = Object.assign({}, this.state.tupleBuffer);
+      updatedBuffer[tableAttribute.attributeName] = tableAttribute.defaulValue; // set to defaulValue for now
+      this.setState({tupleBuffer: updatedBuffer});
+    }
+  }
+
+  /**
+   * Helper function to deal with the creation of the lables to go with the inputs creatd in the getAttributeInputBlock
+   * @param tableAttribute Table attribute object so the function can extract the attributeName (Could add special label/look CSS for differnet types later on)
+   * @param typeString The string to put in () for the user to see of what type the attribute is
+   */
+  getAttributeLabelBlock(tableAttribute: any, typeString: string) {
+    return (
+      <div className="attributeHead">
+        <label style={tableAttribute.constructor === PrimaryTableAttribute ? {color: '#399E5A'} : {color: 'inherit'}} htmlFor={tableAttribute.attributeName}>{tableAttribute.attributeName + ' (' + typeString + ')'}</label>
+        {tableAttribute.constructor === SecondaryTableAttribute && tableAttribute.nullable ? 
+          <div className="nullableControls">
+            <div className="nullableTag">nullable</div>
+            <FontAwesomeIcon className="resetIcon" icon={faRedoAlt} onClick={() => {this.resetToNull(tableAttribute)}} />
+          </div> : ''
+        }
+      </div>
+    );
+  }
```

Comment:
Same thing with the get input block, needs to be extraced out to prevent copying and pasting.